lib/Doctrine/ORM/Tools/Pagination/LimitSubqueryWalker.php

```diff
@@ -66,7 +68,7 @@ public function walkSelectStatement(SelectStatement $AST)
         $pathExpression->type = PathExpression::TYPE_STATE_FIELD;
 
         $AST->selectClause->selectExpressions = [new SelectExpression($pathExpression, '_dctrn_id')];
-        $AST->selectClause->isDistinct        = true;
+        $AST->selectClause->isDistinct        = ($query->getHints()[Paginator::HINT_ENABLE_DISTINCT] ?? null) !== false;
```

Comment:
```suggestion
        $AST->selectClause->isDistinct        = ($query->getHints()[Paginator::HINT_ENABLE_DISTINCT] ?? true) !== false;
```

Makes more sense to make the default more obvious.